planning/behavior_path_planner/autoware_behavior_path_planner_common/test/test_objects_filtering.cpp

```diff
@@ -128,6 +141,108 @@ TEST(BehaviorPathPlanningObjectsFiltering, is_within_circle)
+  EXPECT_TRUE(isPolygonOverlapLanelet(object, toPolygon2d(lanelet)));
+  EXPECT_TRUE(isPolygonOverlapLanelet(object, lanelet, yaw_threshold));
+
+  object.kinematics.initial_pose_with_covariance.pose = createPose(10.0, 10.0, 0.0, 0.0, 0.0, 0.0);
+  EXPECT_FALSE(isPolygonOverlapLanelet(object, lanelet.polygon2d().basicPolygon()));
+  EXPECT_FALSE(isPolygonOverlapLanelet(object, toPolygon2d(lanelet)));
+  EXPECT_FALSE(isPolygonOverlapLanelet(object, lanelet, yaw_threshold));
+}
+
+TEST(BehaviorPathPlanningObjectsFiltering, filterObjects)
+{
+  using autoware::behavior_path_planner::utils::path_safety_checker::filterObjects;
+  using autoware::behavior_path_planner::utils::path_safety_checker::ObjectsFilteringParams;
+  using autoware::universe_utils::createVector3;
+
+  std::shared_ptr<PredictedObjects> objects = std::make_shared<PredictedObjects>();
+  std::shared_ptr<autoware::route_handler::RouteHandler> route_handler =
+    std::make_shared<autoware::route_handler::RouteHandler>();
+  std::shared_ptr<ObjectsFilteringParams> params = std::make_shared<ObjectsFilteringParams>();
+  params->ignore_object_velocity_threshold = false;
+  params->object_check_forward_distance = 20.0;
+  params->object_check_backward_distance = 10.0;
+  params->object_types_to_check.check_car = true;
+  route_handler->setMap(intersection_map);
+  lanelet::ConstLanelets current_lanes;
+
+  current_lanes.push_back(route_handler->getLaneletsFromId(1000));
+  current_lanes.push_back(route_handler->getLaneletsFromId(1010));
+  auto current_pose = createPoint(360.22, 600.51, 0.0);
+
+  EXPECT_TRUE(
+    filterObjects(objects, route_handler, current_lanes, current_pose, params).objects.empty());
+
+  ObjectClassification classification;
+  classification.label = ObjectClassification::Type::CAR;
+  classification.probability = 1.0;
+
+  PredictedObject target_object;
+  target_object.object_id = autoware::universe_utils::generateUUID();
+  target_object.kinematics.initial_pose_with_covariance.pose =
+    createPose(360.22, 605.51, 0.0, 0.0, 0.0, 0.0);
+  target_object.kinematics.initial_twist_with_covariance.twist.linear =
+    createVector3(1.0, 1.0, 0.0);
+  target_object.shape.type = autoware_perception_msgs::msg::Shape::BOUNDING_BOX;
+  target_object.shape.dimensions.x = 1.0;
+  target_object.shape.dimensions.y = 1.0;
+  target_object.classification.push_back(classification);
+
+  PredictedObject other_object;
+  other_object.object_id = autoware::universe_utils::generateUUID();
+  other_object.kinematics.initial_pose_with_covariance.pose =
+    createPose(370.22, 600.51, 0.0, 0.0, 0.0, 0.0);
+  other_object.kinematics.initial_twist_with_covariance.twist.linear = createVector3(1.0, 1.0, 0.0);
+  other_object.shape.type = autoware_perception_msgs::msg::Shape::BOUNDING_BOX;
+  other_object.shape.dimensions.x = 1.0;
+  other_object.shape.dimensions.y = 1.0;
+  other_object.classification.push_back(classification);
```

Comment:
It is possible to create a function that generates bounding box object?

Reply:
Fixed in ebd08455efb8893b4a9ab345c566309ebc35a5d6